src/data/mystery-encounters/utils/encounter-pokemon-utils.ts

```diff
@@ -590,7 +590,7 @@ function failCatch(
   return new Promise<void>((resolve) => {
     globalScene.playSound("se/pb_rel");
     pokemon.setY(originalY);
-    if (pokemon.status?.effect !== StatusEffect.SLEEP) {
+    if (pokemon.hasStatusEffect(StatusEffect.SLEEP)) {
```

Comment:
Previously Comatose was ignored but this make it take Comatose into consideration, I think you want this instead:
```suggestion
    if (pokemon.hasStatusEffect(StatusEffect.SLEEP, false, true)) {
```